app/screens/apps_form/apps_form_field.tsx

```diff
@@ -101,33 +141,45 @@ export default class AppsFormField extends React.PureComponent<Props> {
                     disabled={field.readonly}
                 />
             );
-        } else if (field.type === 'channel' || field.type === 'user' || field.type === 'dynamic_select' || field.type === 'static_select') {
-            let dataSource = ViewTypes.DATA_SOURCE_CHANNELS;
-            if (field.type === 'user') {
+        } else if ([AppFieldTypes.USER, AppFieldTypes.CHANNEL, AppFieldTypes.STATIC_SELECT, AppFieldTypes.DYNAMIC_SELECT].includes(field.type)) {
+            let dataSource = '';
+            let options: DialogOption[] = [];
+
+            switch (field.type) {
+            case AppFieldTypes.USER:
                 dataSource = ViewTypes.DATA_SOURCE_USERS;
+                break;
+            case AppFieldTypes.CHANNEL:
+                dataSource = ViewTypes.DATA_SOURCE_CHANNELS;
+                break;
+            case AppFieldTypes.DYNAMIC_SELECT:
+                dataSource = ViewTypes.DATA_SOURCE_DYNAMIC;
+                break;
+            case AppFieldTypes.STATIC_SELECT:
+                if (field.options) {
+                    options = field.options.map((option) => ({text: option.label, value: option.value}));
```

Comment:
Can we simplify to `field.options?.map(...)`?

Reply:
I'm getting a type error when having it like this:

```
Type '{ text: string; value: string; }[] | undefined' is not assignable to type 'DialogOption[]'.
  Type 'undefined' is not assignable to type 'DialogOption[]'
```

I'll keep it how it is with the if statement